commerce/forms.py

```diff
@@ -213,6 +213,10 @@ def __init__(self, *args, **kwargs):
         self.fields['shipping_option'].queryset = self.instance.shipping_options
 
         payment_method_fieldset = None
+        shipping_option_css = ''
+
+        if not self.instance.shipping_option_selection_required:
```

Comment:
Can we do it by not adding the field to the form instead of using CSS?

Just like payment_method:
```python
payment_method_fieldset = None
if self.instance.billing_details_required:
    payment_method_fieldset = Fieldset(
        _('Choose Payment Type'),
        'payment_method',
        HTML(loader.get_template('commerce/fees.html').render({'fees': self.fields['payment_method'].queryset})),
        css_class='col-md-6'
    )
else:
    self.fields['payment_method'].required = False
```